projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -116,18 +116,26 @@ function jetpack_filter_excerpt_for_newsletter( $excerpt, $post ) {
 }
 
 /**
- * Gate access to posts
+ * Gate access to posts and comments
  *
  * @param string $the_content Post content.
  *
  * @return string
  */
-function maybe_get_locked_content( $the_content ) {
+function maybe_gate_content_and_comments( $the_content ) {
 	require_once JETPACK__PLUGIN_DIR . 'modules/memberships/class-jetpack-memberships.php';
 
 	if ( Jetpack_Memberships::user_can_view_post() ) {
 		return $the_content;
 	}
+
+	// Close comments on the front-end
+	add_filter( 'comments_open', '__return_false' );
```

Comment:
This will get fired in the WordPress loop, attaching a new filter each time a post does this check.
But these filters are not tied to specific post, so:

On a theme that displays comments in a loop for each post, it will block comments for all the posts.

We need to dynamically check these for each post